### ISSUE TITLE
How to use with Sanctum?

### ISSUE THREAD

**Issue by IronSinew** [mll-lab/laravel-graphql-playground#37]
Is it possible to authenticate the playground with sanctum's SPA authentication? Or would it be better to just generate a token and add the bearer auth value to the headers? I've been trying to figure out how to most simply achieve authentication in the playground if a user wanted to test, but transition easily to an external token as well.

**Comment by junioro12** [comment]
Interested in this as well...it seems that the graphql requests don't send the "Referer" header, which Sanctum relies on to authenticate.  If we inject the header in middleware, then it does authenticate, but that isn't secure.  Wondering if there is a way to get the playground graphql calls to send the "Referer" which would seem to solve this.

**Comment by IronSinew** [comment]
I was able to solve this, by the way.

> config/lighthouse.php

change `'gate' => null` to `'gate' => 'sanctum'`

Add `EnsureFrontendRequestsAreStateful::class` to the `route.middleware` config

> config/graphql-playground.php

`route.middleware` I added the `auth` middleware to this, just so that I wouldn't have to get a CSRF cookie / manage login on the playground view. Obviously this could be adjusted to your needs.

Make sure you add this code to `GraphQLPlayground.init` as documented in the other CSRF protection docs:
```
            settings: {
                'request.credentials': 'same-origin',
            },
            headers: {
                'X-CSRF-TOKEN': document.querySelector('meta[name="csrf-token"]').content
            }
```
